cmd/kubeadm/app/util/config/upgradeconfiguration_test.go

```diff
@@ -415,21 +427,24 @@ func TestLoadOrDefaultUpgradeConfiguration(t *testing.T) {
 			},
 		},
 	}
+
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			bytes, err := yaml.Marshal(tt.cfg)
-			if err != nil {
-				t.Fatalf("Could not marshal test config: %v", err)
-			}
-			err = os.WriteFile(filePath, bytes, 0644)
-			if err != nil {
-				t.Fatalf("Couldn't write content to file: %v", err)
-			}
+		for _, format := range formats {
+			t.Run(fmt.Sprintf("%s_%s", tt.name, format.name), func(t *testing.T) {
```

Comment:
since you are adding multi format testing for init and upgrade config, should you also add it to reset and join config?